src/golang/lib/repos/tests/suite.go

```diff
@@ -0,0 +1,100 @@
+package tests
+
+import (
+	"context"
+	"os"
+
+	"github.com/aqueducthq/aqueduct/cmd/migrator/migrator"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/models"
+	"github.com/aqueducthq/aqueduct/lib/repos"
+	"github.com/aqueducthq/aqueduct/lib/repos/sqlite"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+)
+
+// TestSuite contains all fields that are needed by the database integration
+// tests. Each test case should be implemented as a method of TestSuite.
+type TestSuite struct {
+	suite.Suite
+	ctx context.Context
+
+	// List of all repos
+	workflow repos.Workflow
+
+	DB database.Database
+}
+
+// SetupSuite is run only once before all tests. It initializes the database
+// connection and creates the repos. It initializes the database schema
+// to the latest version.
+func (ts *TestSuite) SetupSuite() {
+	DB, err := database.NewSqliteInMemoryDatabase(&database.SqliteConfig{})
+	if err != nil {
+		ts.T().Errorf("Unable to create SQLite client: %v", err)
+	}
+
+	ts.ctx = context.Background()
+	ts.DB = DB
+
+	// Initialize repos
+	ts.workflow = sqlite.NewWorklowRepo()
+
+	// Init database schema
+	if err := initDBSchema(DB); err != nil {
+		DB.Close()
+		ts.T().Errorf("Unable to initialize database schema: %v", err)
+	}
+}
+
+func initDBSchema(DB database.Database) error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// The schema change logic must be invoked from the `golang/` directory
+	if err := os.Chdir("../../.."); err != nil {
+		return err
+	}
+
+	defer func() {
+		chdirErr := os.Chdir(cwd)
+		if chdirErr != nil {
+			log.Errorf("Error when changing cwd: %v", chdirErr)
+		}
+	}()
+
+	if err := migrator.GoTo(context.Background(), models.SchemaVersion, DB); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TearDownTest is run after each test finishes.
+func (ts *TestSuite) TearDownTest() {
+	// Clear all of the tables
+	query := `
```

Comment:
Does this delete everything in the tables? I wonder if there is some way to just delete anything created from the tests..

Reply:
Yeah, this deletes everything, which technically is only things created by the tests since the db is cleared after each test.